lib/client/renderer.js

```diff
@@ -490,30 +621,149 @@ function init (client, d3) {
             'marker-end':'url(#arrow)',
             'x1': chart().xScale(new Date(treatment.mills)),
             'y1': chart().yScale(client.sbx.scaleEntry(treatment)),
-            'x2': d3.event.x - this.offsetWidth/2,
-            'y2': chart().yScale(client.sbx.scaleEntry(treatment)),
+            'x2': x,
+            'y2': y,
             'stroke-width': 2,
             'stroke': 'white'
           });
+          
       })
       .on('dragend', function() { 
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1988570277/issues/source?bid=2452379&fileBranchId=3084282#l631)